apps/system/js/system_window.js

```diff
@@ -0,0 +1,51 @@
+/* global AudioChannelController */
+'use strict';
+
+(function(exports) {
+  function SystemWindow() {
+    this.audioChannels = new Map();
+    window.addEventListener('mozChromeEvent', this);
+    // Get System app's audio channels.
+    this._sendContentEvent({ type: 'system-audiochannel-list' });
+  }
+
+  SystemWindow.prototype = {
+    audioChannels: null,
+
+    /**
+     * General event handler interface.
+     *
+     * @param {Event} evt The event to handle.
+     */
+    handleEvent: function(evt) {
+      switch (evt.type) {
+        case 'mozChromeEvent':
+          var detail = evt.detail;
+          switch (detail.type) {
+            case 'system-audiochannel-list':
+              detail.audioChannels.forEach((name) => {
+                this.audioChannels.set(
+                  name, new AudioChannelController(name)
```

Comment:
I think you meant we could do `new AudioChannelController(this, name);`.
The comment[1] is good idea. Learned it.
[1]: https://bugzilla.mozilla.org/show_bug.cgi?id=1157140#c31